x-pack/plugins/security/server/audit/audit_service.ts

```diff
@@ -5,53 +5,176 @@
+import { SpacesPluginSetup } from '../../../spaces/server';
+import { AuditEvent, httpRequestEvent } from './audit_events';
+import { SecurityPluginSetup } from '..';
 
-export interface AuditLogger {
+/**
+ * @deprecated
+ */
+export interface LegacyAuditLogger {
   log: (eventType: string, message: string, data?: Record<string, any>) => void;
 }
 
+export interface AuditLogger {
+  log: (event: AuditEvent) => void;
+}
+
+interface AuditLogMeta extends AuditEvent {
+  session?: {
+    id: string;
+  };
+  trace: {
+    id: string;
+  };
+}
+
 export interface AuditServiceSetup {
-  getLogger: (id?: string) => AuditLogger;
+  withRequest: (request: KibanaRequest) => AuditLogger;
+  getLogger: (id?: string) => LegacyAuditLogger;
 }
 
 interface AuditServiceSetupParams {
   license: SecurityLicense;
   config: ConfigType['audit'];
+  logging: Pick<LoggingServiceSetup, 'configure'>;
+  http: Pick<HttpServiceSetup, 'registerOnPostAuth'>;
+  getCurrentUser(
+    request: KibanaRequest
+  ): ReturnType<SecurityPluginSetup['authc']['getCurrentUser']> | undefined;
+  getSpaceId(
+    request: KibanaRequest
+  ): ReturnType<SpacesPluginSetup['spacesService']['getSpaceId']> | undefined;
 }
 
 export class AuditService {
+  /**
+   * @deprecated
+   */
   private licenseFeaturesSubscription?: Subscription;
-  private auditLoggingEnabled = false;
+  /**
+   * @deprecated
+   */
+  private allowAuditLogging = false;
+  private readonly betaLogger: Logger;
 
-  constructor(private readonly logger: Logger) {}
+  constructor(private readonly logger: Logger) {
+    this.betaLogger = this.logger.get('beta');
+  }
 
-  setup({ license, config }: AuditServiceSetupParams): AuditServiceSetup {
-    if (config.enabled) {
+  setup({
+    license,
+    config,
+    logging,
+    http,
+    getCurrentUser,
+    getSpaceId,
+  }: AuditServiceSetupParams): AuditServiceSetup {
+    if (config.enabled && !config.appender) {
       this.licenseFeaturesSubscription = license.features$.subscribe(({ allowAuditLogging }) => {
-        this.auditLoggingEnabled = allowAuditLogging;
+        this.allowAuditLogging = allowAuditLogging;
       });
     }
 
-    return {
-      getLogger: (id?: string): AuditLogger => {
-        return {
-          log: (eventType: string, message: string, data?: Record<string, any>) => {
-            if (!this.auditLoggingEnabled) {
-              return;
-            }
-
-            this.logger.info(message, {
-              tags: id ? [id, eventType] : [eventType],
-              eventType,
-              ...data,
-            });
+    // Configure logging
+    logging.configure(license.features$.pipe(createLoggingConfig(config)));
+
+    /**
+     * Creates an {@link AuditLogger} scoped to the current request.
+     *
+     * @example
+     * ```typescript
+     * const auditLogger = securitySetup.audit.withRequest(request);
+     * auditLogger.log(event);
+     * ```
+     */
+    const withRequest = (request: KibanaRequest): AuditLogger => {
```

Comment:
Optional: in the platform, all such fabrics are named `asScoped`. We might consider following the same convention. even in the js docs it states 'Creates an {@link AuditLogger} **scoped** to the current request.

Reply:
Nice, I'll follow that convention then. I've seen "with request" style higher order functions too but happy to align with core here for consistency